apps/prairielearn/src/lib/editors.ts

```diff
@@ -983,13 +985,39 @@ export class CourseInstanceCopyEditor extends Editor {
       // making the copied course instance available to users.
       infoJson['allowAccess'] = [];
 
+      const questionsForCopy = await selectQuestionsForCourseInstanceCopy(this.course_instance.id);
+      const questionsToImport = new Set(
+        questionsForCopy.filter((q) => !q.should_copy).map((q) => q.qid),
+      );
+
+      const questionsToCopy = new Set(questionsForCopy.filter((q) => q.should_copy));
+      const newQids = {};
+      for (const question of questionsToCopy) {
+        if (!question.qid) {
+          continue; // TODO better error handling?
+        }
```

Comment:
I'm not sure what the correct error handling is here, its kinda weird that `qid` is nullable.

Reply:
Changed to `assert(...)` in 8365509.